python/tests/test_account.py

```diff
@@ -1450,24 +1451,22 @@ def test_set_get_group_image(self, acfactory, data, lp):
         assert chat.is_promoted()
 
         lp.sec("ac2: wait for receiving message from ac1")
-        ev = ac2._evtracker.get_matching("DC_EVENT_INCOMING_MSG")
-        msg_in = ac2.get_message_by_id(ev.data2)
+        msg_in = ac2._evtracker.wait_next_incoming_message()
         assert not msg_in.chat.is_deaddrop()
+        msg_in.text == "hi"
 
-        lp.sec("ac2: create chat and read profile image")
-        chat2 = ac2.create_chat_by_message(msg_in)
-        p2 = chat2.get_profile_image()
+        lp.sec("ac2: see if chat now has got the profile image")
+        ac2_chat = ac2.create_chat_by_message(msg_in)
+        p2 = ac2_chat.get_profile_image()
         assert p2 is not None
         assert open(p2, "rb").read() == open(p, "rb").read()
 
         ac2._evtracker.consume_events()
         ac1._evtracker.consume_events()
         lp.sec("ac2: delete profile image from chat")
-        chat2.remove_profile_image()
-        ev = ac1._evtracker.get_matching("DC_EVENT_INCOMING_MSG")
-        assert ev.data1 == chat.id
-        chat1b = ac1.create_chat_by_message(ev.data2)
-        assert chat1b.get_profile_image() is None
+        msg_in.chat.remove_profile_image()
+        msg_back = ac1._evtracker.wait_next_incoming_message()
+        assert msg_back.chat == chat
```

Comment:
I guess you know what you are doing here

Reply:
hope so as well

Reply:
it's just a little clenaup